### PR TITLE
Feature/redesign reference

### PR DESCRIPTION
**Pull request by kzalite**

### CODE REVIEW

File: app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/InputCell/DistributionCellDialog/DistributionInputFields/DistributionInputFields.tsx [hunk 1]
```diff
@@ -35,7 +36,7 @@ export default function DistributionInputFields({
   }
 
   function handleKeyDown(event: KeyboardEvent<HTMLDivElement>): void {
-    if (event.keyCode === 13 && !isInputInvalid()) {
+    if (event.keyCode === keycode('enter') && !isInputInvalid()) {
```
Comment (DanielReid) [inline app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/InputCell/DistributionCellDialog/DistributionInputFields/DistributionInputFields.tsx:39]: Excellent. Did you check the rest of the code as well for stray instances by any chance?

File: app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx [hunk 2]
```diff
@@ -15,25 +16,62 @@ export default function ReferenceCell({
...
+      referenceLink: newReferenceLink
+    });
+    closeDialog();
+  }
+
+  function renderReference(): JSX.Element {
+    if (
+      dataSource.reference &&
+      dataSource.referenceLink &&
+      !checkIfLinkIsInvalid(dataSource.referenceLink)
```
Comment (DanielReid) [inline app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx:36]: `isLinkValid` ?

File: app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx [hunk 3]
```diff
@@ -15,25 +16,62 @@ export default function ReferenceCell({
...
+    ) {
+      return (
+        <Tooltip title={tooltipText}>
+          <>
+            <div>{dataSource.reference}</div>
+            <div>({dataSource.referenceLink})</div>
```
Comment (DanielReid) [inline app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx:42]: why not show the referencelink if it's invalid? it seems more informative to show it but maybe do markup if it's invalid?

File: test/endToEnd/manualInputTest.js [hunk 4]
```diff
@@ -445,37 +435,32 @@ function editReference(browser) {
...
-        .containsText(referenceLink, 'click to edit')
-        .click(referenceLink)
-        .setValue(referenceLinkInput, 'not_a_link')
-        .click('//*[@id="favourable-criteria-label"]')
-        .assert.containsText(referenceLink, INVDALID_REFERENCE_LINK_WARNING)
-        .click(referenceLink)
+      browser.click(basePath).setValue(referenceLinkInput, 'not_a_link');
+      browser.expect
+        .element('//*[@id="reference-error-0"]')
+        .text.to.equal(INVDALID_REFERENCE_LINK_WARNING);
```
Comment (DanielReid) [inline test/endToEnd/manualInputTest.js:456]: such invdalid

**Review by DanielReid — CHANGES_REQUESTED** [review]
nice; bit funny situation with invalid links

File: app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx (hunk 3 above)
Comment (kzalite) [reply app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx:42]: Can't enter invalid link, i.e. can't commit the changes in reference dialog

**Review by chinook25 — APPROVED** [review]

**Pull request merged**